### ISSUE TITLE
Loading issue on iPad Pro only ("request desktop site") 

### ISSUE THREAD

**Issue by taitems** [CookPete/react-player#1004]
**TL;DR** - The isIOS check is out of date and doesn't handle iPad Pro in desktop mode.

#### Current Behavior

If you navigate to a video shown by react-player on an iPad Pro specifically, it never passes `loading` to fire the `onReady` event. Looking at the network tab you will see a red `206 Partial Content` instead of the normal green `206`.

In our use case where we show a spinner tied to `loading`, it never goes away.

By default the iPad Pro is requesting the **desktop site**.

- iPad Pro “request desktop site” user agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.5 Safari/605.1.15
- iPad Pro “request mobile site” user agent: Mozilla/5.0 (iPad; CPU OS 13_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.5 Mobile/15E148 Safari/604.1

#### Expected Behavior
Video loads but still needs to be clicked to play, as other issues have noted due to changing iOS autoplay settings.

#### Steps to Reproduce

1. Navigate to a page in an iPad Pro
1. Ensure it is in "desktop mode"
1. Video will not load
1. "Request mobile mode"
1. Note the video will now loads successfully

#### Environment
- Behind an auth wall unfortunately.

#### Other Information
- Creating a PR to fix this by implementing the improved isIOS check from here https://stackoverflow.com/a/58065241
